src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1726,12 +1726,12 @@ int SR_initialize() {
   char *s;
   // Get signal number to use for suspend/resume
   if ((s = ::getenv("_JAVA_SR_SIGNUM")) != 0) {
-    int sig = ::strtol(s, 0, 10);
+    int64_t sig = ::strtol(s, 0, 10);
     if (sig > MAX2(SIGSEGV, SIGBUS) &&  // See 4355769.
         sig < NSIG) {                   // Must be legal signal and fit into sigflags[].
-      PosixSignals::SR_signum = sig;
+      PosixSignals::SR_signum = checked_cast<int>(sig);
```

Comment:
checked cast seems unnecessary given the range check

Reply:
I agree but I added it to make other reviewers happy.

Reply:
An alternative would be to use parse_integer() instead of strtol to get the value as an int.

Reply:
parse_integer is nicer so I did that.